db/migrate/20190829102315_http_proxy_unique_name.rb

```diff
@@ -0,0 +1,20 @@
+class HttpProxyUniqueName < ActiveRecord::Migration[5.2]
+  def up
+    names = HttpProxy.group(:name).count.select { |key, value| value > 1 }.keys
+    unless names.empty?
+      names.each do |name|
+        proxies = HttpProxy.where :name => name
+        say "#{proxies.count} Http Proxies with duplicate names detected: #{name}"
+      end
+      raise "Please remove any duplicate Http Proxy records from the DB before continuing."
+    end
+
```

Comment:
@tbrisker Made changes to handle existing duplicate records in DB. Although, could you guide me here, if raising a msg asking user to delete the records accordingly (as above) or, dropping the record in our migration itself, what should be a better choice for http proxies?

Reply:
I think ideally we shouldn't stop the migration if possible since that will make it difficult for the user to come back from. Perhaps instead rename the duplicates and print a message indicating that they were renamed?

Reply:
@tbrisker As per suggested changes, I've tried here to rename the record by appending it's **id** in the end of the name to avoid any differences in the naming of every record.
Thoughts?
Also, working on related tests now.

Reply:
you can use something like:
```ruby
duplicates = HttpProxy.group(:name).count.select{|k,v| v > 1 }.keys
HttpProxy.where(name: duplicates).find_each do |proxy|
  ....
end
```
instead of iterating over all of them and maintaining a list of names

Reply:
done